src/pages/Home/Home.jsx

```diff
@@ -6,6 +6,31 @@ import { Link } from "react-router-dom";
 import axios from "axios";
 
 export const Home = () => {
+  const [categories, setCategories] = useState([]);
+  const [homeProducts, setHomeProducts] = useState([]);
```

Comment:
we should create a ProductContext or DataContext with reducers. you need products & categories on the product listing page as well. I would suggest using useState for local state only.

Reply:
But I am using those data on that page only.

Reply:
Maybe you will need categories in more places as well. So might as well move it to context.